src/main/java/com/microsoft/graph/content/MSBatchRequestContent.java

```diff
@@ -5,145 +5,152 @@
 			removed = true;
-			for(Map.Entry<String, MSBatchRequestStep> steps : batchRequestStepsHashMap.entrySet()) {
-				if(steps.getValue() != null && steps.getValue().getArrayOfDependsOnIds() != null) {
-					while(steps.getValue().getArrayOfDependsOnIds().remove(requestId));
+			for (final Map.Entry<String, MSBatchRequestStep> steps : batchRequestStepsHashMap.entrySet()) {
+				if (steps.getValue() != null && steps.getValue().getArrayOfDependsOnIds() != null) {
+					while (steps.getValue().getArrayOfDependsOnIds().remove(requestId))
+						;
 				}
 			}
 		}
 		return removed;
 	}
-	
+
 	/*
 	 * @return Batch request content's json as String
 	 */
 	public String getBatchRequestContent() {
-		JSONObject batchRequestContentMap = new JSONObject();
-		JSONArray batchContentArray = new JSONArray();
-		for(Map.Entry<String, MSBatchRequestStep> requestStep : batchRequestStepsHashMap.entrySet()) {
+		final JsonObject batchRequestContentMap = new JsonObject();
+		final JsonArray batchContentArray = new JsonArray();
+		for (final Map.Entry<String, MSBatchRequestStep> requestStep : batchRequestStepsHashMap.entrySet()) {
 			batchContentArray.add(getBatchRequestObjectFromRequestStep(requestStep.getValue()));
 		}
-		batchRequestContentMap.put("requests", batchContentArray);
-		
-		String content = batchRequestContentMap.toString();
+		batchRequestContentMap.add("requests", batchContentArray);
+
+		final String content = batchRequestContentMap.toString();
 		return content;
 	}
-	
-	private JSONObject getBatchRequestObjectFromRequestStep(final MSBatchRequestStep batchRequestStep){
-		JSONObject contentmap = new JSONObject();
-		contentmap.put("id", batchRequestStep.getRequestId());
-		
-		String url = batchRequestStep.getRequest().url().toString();
-		url = url.replaceAll("https://graph.microsoft.com/v1.0/", "");
-		url = url.replaceAll("http://graph.microsoft.com/v1.0/", "");
-		url = url.replaceAll("https://graph.microsoft.com/beta/", "");
-		url = url.replaceAll("http://graph.microsoft.com/beta/", "");
-		contentmap.put("url", url);
-		
-		contentmap.put("method", batchRequestStep.getRequest().method().toString());
-		
-		Headers headers = batchRequestStep.getRequest().headers();
-		if(headers != null && headers.size() != 0) {
-			JSONObject headerMap = new JSONObject();
-			for(Map.Entry<String, List<String>> entry : headers.toMultimap().entrySet()) {
-				headerMap.put(entry.getKey(), getHeaderValuesAsString(entry.getValue()));
+
+	private JsonObject getBatchRequestObjectFromRequestStep(final MSBatchRequestStep batchRequestStep) {
+		final JsonObject contentmap = new JsonObject();
+		contentmap.add("id", new JsonPrimitive(batchRequestStep.getRequestId()));
+
+		final String url = batchRequestStep.getRequest().url().toString()
+				.replaceAll("https://graph.microsoft.com/v1.0/", "").replaceAll("http://graph.microsoft.com/v1.0/", "")
+				.replaceAll("https://graph.microsoft.com/beta/", "").replaceAll("http://graph.microsoft.com/beta/", "");
```

Comment:
Wouldn't this be better as a single, pre-compilled regex (you can just group the part that changes.  moreover there are regex chars in there that are not escaped like `.`) instead of re-compiling 4 separate regexes every time this method is run?

Reply:
yes it would.
However in this PR I was simply looking at replacing simple-json and having the lowest changes possible on the codebase, hence why I didn't change it.

Reply:
https://github.com/microsoftgraph/msgraph-sdk-java-core/issues/66